make error in case of './configure --enable-trace' because of '-nopie' option

```
$ ./autogen.sh
[..]

$ ./configure --disable-corosync --enable-trace
[..]
  Trace        CFLAGS      = -pg -nopie
  Final        CFLAGS      = -g -O2 -DNDEBUG -ggdb3  	-pg -nopie   -Wall -Wshadow -Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes -Wpointer-arith -Wwrite-strings -Wbad-function-cast -Wmissing-format-attribute -Wformat=2 -Wformat-security -Wformat-nonliteral -Wno-long-long -Wno-strict-aliasing -Wcast-align  	  -D_GNU_SOURCE -D_LGPL_SOURCE 	-std=gnu99
  Final        CPPFLAGS    =   
  Final        LDFLAGS     =   

$ make
Making all in lib
make[1]: Entering directory '/home/sheep/sheepdog/lib'
  CC       shared/libsheepdog_a-sheep.o
gcc: error: unrecognized command line option '-nopie'
Makefile:704: recipe for target 'shared/libsheepdog_a-sheep.o' failed
make[1]: *** [shared/libsheepdog_a-sheep.o] Error 1
make[1]: Leaving directory '/home/sheep/sheepdog/lib'
Makefile:496: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1

$ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

I think gcc does not have `-nopie` option but has `-f[no-]pie` [code generation option](https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html) or `-[no-]pie` [link option](https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html).

--- a/configure.ac
+++ b/configure.ac
@@ -541,7 +541,7 @@ fi
 if test "x${enable_trace}" = xyes && \
 		cc_supports_flag -pg ; then
 	AC_MSG_NOTICE([Enabling trace (-pg)])
-	TRACE_CFLAGS="-pg -nopie"
+	TRACE_CFLAGS="-pg -fno-pie"
 else
 	TRACE_CFLAGS=""
 fi

